### DISCUSSION TITLE
block size of store object

### DISCUSSION THREAD

**Discussion by gihyeonjeon** [gem5/gem5#818] [Q&A] [answered]
hi, I'm studying about gem5.
And I know that the unit of cache storage is cache Line Size.

But in my case, I want make a new store object which stores data from main memory as its packet size, i mean that variable size.
For that, I have copied the simple cache code, and have modified.
I have modified the code for sending request packets from the data storage space I created to the main memory to request only the size I desire. Additionally, I revised the insert code to store data in the designated storage space according to the size of the response packet received from memory. I also adjusted the code so that if there is data to be deleted within the insert function, it will write back to the main memory only the size of that data. In other words, I have modified it so that the cache line size in both requests and responses is processed only as much as needed by the CPU from my created memory space. However, it is not functioning correctly. There is a page fault occurring, and upon using gdb, it was found that the vaddr could not be found in the TLB lookup, leading to the error. The first possible cause of this issue is that the TLB was created incorrectly, and the second is that an incorrect vaddr was referenced when referencing the vaddr. Therefore, I plan to start by checking the code that initializes the vaddr.

I would like to know if there is a specific part I should look at to solve this problem.
I think that this created memory space is less efficient, but this is intentional.

Thank you!

**Comment by powerjg** [comment, marked as answer]
My bet is that you're getting a functionally incorrect value out of the cache. I would use the `Exec` debug flag to get a trace of all instructions and compare the trace from the old cache and your new design. I expect the trace will differ at some point, and that will help you pinpoint the bug.

**Reply by gihyeonjeon** [reply]
Thanks!..
I have solved this problem using some debug flags and gdb..
